pkg/routeagent/controllers/route/iptables.go

```diff
@@ -95,3 +96,43 @@ func (r *Controller) programIptableRulesForInterClusterTraffic(remoteCidrBlock s
 
 	return nil
 }
+
+func (r *Controller) clearGlobalnetChains() error {
+	if r.globalnetStatus == GN_Status_Not_Verified {
```

Comment:
The code seems in general good, but I propose an improvement.

Here we are moving the cleanup logic back into the route agent, which is what we want. But remember that we have technical debt still for this and ipsec too that we wanted to cleanup. We are adding more.

Why don't we design a simple interface for cleanups, and have a cleanup object in globalnet of which we can create an instance here,and just call it at the right hook points.

That way, we could later do the same for the different cable drivers. We'd load the "cleanup" interface of the cable driver, and use it.

We could even have a list of those objects, and when we call the hook for cleanup all execute. We could have a single interface with multiple events (of when a cleanup could need to happen...
- GatewayStarted, GatewayStopped, RouteAgentStopped, stuff like that. (I'm enumerating some random ones which may be make no sense).

Reply:
Thanks for your review comments @mangelajo. I've updated the code, please take a look.